generator/test/test_util.dart

```diff
@@ -180,36 +179,27 @@ class LinesEqualsMatcher extends Matcher {
 Future<Map<String, Object>> getExpectedFileAsset(String inputFilePath, String expectedFilePath) async {
   var key = 'pkg|${_path.setExtension(inputFilePath, '.realm_objects.g.part')}';
   return {
-    key: GoldenFileMatcher(
-      File(expectedFilePath),
-      (f) => MyersDiffMatcher(DartFormatter(lineEnding: '\n').format(f.readAsStringSync())),
-    )
+    key: myersDiff(await File(expectedFilePath).readFileAsDartCode()),
   };
 }
 
-Future<String> readFileAsDartFormattedString(String path) async {
-  var file = File(_path.join(Directory.current.path, path));
-  String content = await file.readAsString(encoding: utf8);
-  return _stringReplacements(content);
-}
-
-Future<String> readFileAsErrorFormattedString(String directoryName, String outputFilePath) async {
-  var file = File(_path.join(Directory.current.path, '$directoryName/$outputFilePath'));
-  String content = await file.readAsString(encoding: utf8);
-  if (Platform.isWindows) {
-    var macToWinSymbols = {'╷': ',', '━': '=', '╵': '\'', '│': '|', '─': '-', '┌': ',', '└': '\''};
-    macToWinSymbols.forEach((key, value) {
-      content = content.replaceAll(key, value);
-    });
+extension FileEx on File {
+  Future<String> readFileAsErrorFormattedString() async {
+    var content = await readAsString(encoding: utf8);
+    if (Platform.isWindows) {
+      var macToWinSymbols = {'╷': ',', '━': '=', '╵': '\'', '│': '|', '─': '-', '┌': ',', '└': '\''};
+      macToWinSymbols.forEach((key, value) {
+        content = content.replaceAll(key, value);
+      });
+    }
+    return LineSplitter.split(content).join('\n');
   }
-  return LineSplitter.split(content).join('\n');
-}
 
-String _stringReplacements(String content) {
-  final formatter = DartFormatter();
-  var lines = LineSplitter.split(content);
-  String formattedContent = lines.where((element) => !element.startsWith("part of")).join('\n');
-  return formatter.format(formattedContent);
+  Future<String> readFileAsDartCode() async {
```

Comment:
I added a suggestion to change the name of the method, because it could be used also for formatting any string content, not only from file.

Reply:
This is named to follow convention. LIke `readFileAsString`.